controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -253,6 +261,89 @@ func (r *HostedControlPlaneReconciler) isVpcEndpointReady(ctx context.Context, h
 	}
 }
 
+// This will install another RouteMonitor for the external FQDN of the apiserver.
+// We also install Dynatrace probes for this, but those might get replaced, so this is ensuring alerting is maintained.
+func (r *HostedControlPlaneReconciler) deployExternalMonitoringObjects(ctx context.Context, log logr.Logger, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) error {
+	// The Route we want to monitor should already exist
+	apiserverRoute := routev1.Route{}
+	err := r.Get(ctx, types.NamespacedName{
+		Namespace: hostedcontrolplane.Namespace,
+		Name:      "kube-apiserver",
+	}, &apiserverRoute)
+	if err != nil {
+		log.Error(err, "failed to retrieve 'kube-apiserver' route - can not setup routemonitor")
+		return err
+	}
+	apiserverServiceName := apiserverRoute.Spec.To.Name
+	if apiserverServiceName == "" {
+		return fmt.Errorf("kube-apiserver route is not referring to a service in 'Spec.To.Name'")
```

Comment:
For a cluster with it's API Listing set to `internal` there will not be a `kube-apiserver` route by default (it's named `kube-apiserver-internal`). Can we maybe check if a cluster is internal before calling `deployExternalMonitoringObjects()` and skip the check when internal?